src/lib/mdns/Advertiser.h

```diff
@@ -179,34 +179,37 @@ class CommissionAdvertisingParameters : public BaseAdvertisingParams<CommissionA
     {
         if (deviceName.HasValue())
         {
-            strncpy(sDeviceName, deviceName.Value(), min(strlen(deviceName.Value()) + 1, sizeof(sDeviceName)));
-            mDeviceName = Optional<const char *>::Value(static_cast<const char *>(sDeviceName));
+            strncpy(mDeviceName, deviceName.Value(), min(strlen(deviceName.Value()) + 1, sizeof(mDeviceName)));
+            mDeviceNameHasValue = true;
         }
         return *this;
     }
-    Optional<const char *> GetDeviceName() const { return mDeviceName; }
+    bool HasValueDeviceName() const { return mDeviceNameHasValue; }
+    const char * GetDeviceName() const { return mDeviceName; }
 
     CommissionAdvertisingParameters & SetRotatingId(Optional<const char *> rotatingId)
     {
         if (rotatingId.HasValue())
         {
-            strncpy(sRotatingId, rotatingId.Value(), min(strlen(rotatingId.Value()) + 1, sizeof(sRotatingId)));
-            mRotatingId = Optional<const char *>::Value(static_cast<const char *>(sRotatingId));
+            strncpy(mRotatingId, rotatingId.Value(), min(strlen(rotatingId.Value()) + 1, sizeof(mRotatingId)));
+            mRotatingIdHasValue = true;
         }
         return *this;
     }
-    Optional<const char *> GetRotatingId() const { return mRotatingId; }
+    bool HasValueRotatingId() const { return mRotatingIdHasValue; }
+    const char * GetRotatingId() const { return mRotatingId; }
 
     CommissionAdvertisingParameters & SetPairingInstr(Optional<const char *> pairingInstr)
     {
         if (pairingInstr.HasValue())
         {
-            strncpy(sPairingInstr, pairingInstr.Value(), min(strlen(pairingInstr.Value()) + 1, sizeof(sPairingInstr)));
-            mPairingInstr = Optional<const char *>::Value(static_cast<const char *>(sPairingInstr));
+            strncpy(mPairingInstr, pairingInstr.Value(), min(strlen(pairingInstr.Value()) + 1, sizeof(mPairingInstr)));
+            mPairingInstrHasValue = true;
```

Comment:
if it does not have value, you need an else branch here and in the other places to set 'has' to false.